### ISSUE TITLE
Downgrading Redis engine version should force new resource

### ISSUE THREAD

**Issue by jurajseffer** [terraform-providers/terraform-provider-aws#2421]
### Terraform Version
Terraform v0.10.8
AWS provider v1.3.0

### Affected Resource(s)
`aws_elasticache_cluster`

### Terraform Configuration Files
```hcl
resource "aws_elasticache_cluster" "instance" {
  engine = "redis"
  engine_version = "3.2.10"
  ...
```

```hcl
resource "aws_elasticache_cluster" "instance" {
  engine = "redis"
  engine_version = "3.2.4"
  ...
```
### Expected Behavior
Plan should show the resource needs to be destroyed and recreated, not updated in place.

### Actual Behavior
Terraform plan shows update in place planned.
```
~ module.redis.aws_elasticache_cluster.instance
      engine_version:     "3.2.10" => "3.2.4"
```
and apply fails because AWS doesn't seem to allow updating a running node like that.
```
aws_elasticache_cluster.instance: [WARN] Error updating ElastiCache cluster (***), error: InvalidParameterCombination: Cannot modify redis from 3.2.10 to 3.2.4
```
> Important, You can upgrade to newer engine versions (see Upgrading Engine Versions), but you cannot dwngrade to older engine versions except by deleting the existing cluster and creating it anew.

http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/Clusters.Modify.html

**Comment by apparentlymart** [comment]
Hi @jurajseffer! Thanks for reporting this bug.

Until recently it was not possible for a provider to dynamically "force new resource" to handle cases where e.g. it's required only when _downgrading_, but this is now possible thanks to the `CustomizeDiff` facility that allows a provider to specify custom logic to run to make dynamic adjustments to the diff. As long as there is a well-defined way to decide whether a version is older or newer than another, it should be possible to have the provider mark this attribute as requiring a new resource if the new value is older than the previous value, using [`ForceNewIfChange`](https://godoc.org/github.com/hashicorp/terraform/helper/customdiff#ForceNewIfChange).

**Comment by bflad** [comment]
This has been implemented in #3857, merged into master, and will release with v1.12.0 of the AWS provider, likely at the end of this week. Happy Terraform'ing! 🎉

**Comment by bflad** [comment]
This has been released in [version 1.12.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1120-march-23-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
